### PR TITLE
source/pg: retrieve and log our backend PID and wal_sender_timeout

### PR DESCRIPTION
**Pull request by petrosagg**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/storage/src/source/postgres/replication.rs [hunk 1]
```diff
@@ -209,10 +209,17 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
...
+                    &config.config.connection_context.ssh_tunnel_manager,
+                )
+                .await?;
+
             tracing::info!(%id, "ensuring replication slot {slot} exists");
             super::ensure_replication_slot(&replication_client, slot).await?;
-            let slot_lsn = super::fetch_slot_resume_lsn(
-                &replication_client,
+            let slot_metadata = super::fetch_slot_metadata(
+                &metadata_client,
```
Comment (rjobanp) [inline src/storage/src/source/postgres/replication.rs:222]: why was it important to switch the client/connection being used here? dont we also want to log the PID of the replication client?

**Review by rjobanp — APPROVED** [review]
curious about one thing but otherwise LGTM

File: src/storage/src/source/postgres.rs [hunk 2]
```diff
@@ -349,25 +349,39 @@ async fn ensure_replication_slot(client: &Client, slot: &str) -> Result<(), Tran
     }
 }
 
+/// The state of a replication slot.
+struct SlotMetadata {
+    /// The process ID of the session using this slot if the slot is currently actively being used.
+    /// None if inactive.
+    active_pid: Option<i32>,
```
Comment (benesch) [inline src/storage/src/source/postgres.rs:356]: I'm confused about this! I was thinking we'd emit the backend PID for _our_ connection, but this is emitting the backend PID, if any, for the connection that was using the slot previously? I _think_ that tends to already get printed to the logs when the "replication slot active for PID" error gets hit later when we try to claim the replication slot.

Comment (benesch) [reply src/storage/src/source/postgres.rs:356]: Ugh, tokio postgres makes this hard to get at. Working on it here: https://github.com/MaterializeInc/rust-postgres/pull/24

File: src/storage/src/source/postgres/replication.rs (hunk 1 above)
Comment (petrosagg) [reply src/storage/src/source/postgres/replication.rs:222]: It wasn't important to switch the connection, I just wanted to use the nicer type aware API that is only available on the normal client and not the simple query only API that is available in the replication client.

**Review by petrosagg — COMMENTED** [review]
Nikhil's commit LGTM

**Pull request merged**